layer/trace_layer.cpp

```diff
@@ -98,6 +98,32 @@ static VkInstance get_instance_handle(const void* handle)
     return (entry != instance_handles.end()) ? entry->second : VK_NULL_HANDLE;
 }
 
+// The vk_layerGetPhysicalDeviceProcAddr of the next layer in the chain.
+// Retrieved during instance creation and forwarded to by this layer's
+// GetPhysicalDeviceProcAddr() after unwrapping its VkInstance parameter.
+static std::mutex                                                    gpdpa_lock;
+static std::unordered_map<VkInstance, PFN_GetPhysicalDeviceProcAddr> next_gpdpa;
+
+static void set_next_gpdpa(const VkInstance instance, PFN_GetPhysicalDeviceProcAddr p_next_gpdpa)
```

Comment:
My preference here, if it's okay with you, is to also mention "instance" in the function name (and below for "get") to echo that it's keyed from the instance, like `get_instance_handle` above.  Maybe `{get,set}_instance_next_gpdpa`?